src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -163,6 +165,49 @@ public void Startup(StartupParams startupParams)
             noNetworkMode = startupParams.NoNetworkMode;
         }
 
+        /// <summary>
+        /// When a new engine is added from a package, add its dependency to the respective package in the dictionary.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        internal void PythonEngineAdded(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                try
+                {
+                    var assem = e.NewItems[0]?.GetType().Assembly;
+                    if (assem == null) return;
+
+                    var assemLoc = assem.Location;
+                    foreach (var pkg in PackageLoader.LocalPackages)
+                    {
+                        if (assemLoc.StartsWith(pkg.RootDirectory))
+                        {
+                            if (NodePackageDictionary.ContainsKey(assem.FullName))
+                            {
+                                var assemName = AssemblyName.GetAssemblyName(assem.Location);
+                                OnMessageLogged(LogMessage.Info(
+                                    string.Format("{0} contains the python engine library {1}, which has already been loaded " +
+                                    "by another package. This may cause inconsistent results when determining which " +
+                                    "python engine the nodes are dependent on.", pkg.Name, assemName.Name)
+                                    ));
+                            }
+                            else
+                            {
+                                NodePackageDictionary[assem.FullName] = new List<PackageInfo>();
+                            }
+                            NodePackageDictionary[assem.FullName].Add(new PackageInfo(pkg.Name, new Version(pkg.VersionName)));
```

Comment:
pkg.VersionName is a version, or a name, or what?

Reply:
It is the version string of the package - "0.0.1" etc